mlir/lib/Tools/PDLL/Parser/Parser.cpp

```diff
@@ -1915,28 +1964,180 @@ FailureOr<ast::Expr *> Parser::parseEqualityExpr() {
 FailureOr<ast::Expr *> Parser::parseRelationExpr() { return parseAddSubExpr(); }
 
 FailureOr<ast::Expr *> Parser::parseAddSubExpr() {
-  auto lhs = parseMulDivExpr();
+  auto lhs = parseMulDivModExpr();
+  if (failed(lhs))
+    return failure();
+
+  for (;;) {
```

Comment:
Why does this need a loop?

Reply:
It enables multiple operations of same precedence, e.g. `let a : Attr = attr<"4 : i32"> + attr<"5 : i32"> - attr<"6 : i32">`. Does it make sense to you?

Reply:
I think you can achieve the same by doing `rhs = parseAddSubExpr()` below,
which, from what I understand, is the typical way how recursive descent parsers are implemented.
Would that work?